### PR TITLE
Feature/v haiboz

### PR DESCRIPTION
**Pull request by v-haiboz**

### CODE REVIEW

File: ECMA2Yaml/ECMAHelper/Metadata/F1KeywordsGenerator.cs [hunk 1]
```diff
@@ -44,6 +44,84 @@ private static IEnumerable<string> GetF1Keywords(ReflectionItem item)
...
+            if (lastindex != -1)
+                yield return uid.Substring(lastindex + 1);
+
+            /*for member page
+             *Current keywords(member page): System.Collections.Generic.Dictionary.Keycollection.Enumerator
+             * Keywords to add: Keycollection.Enumerator
+             *                  Keycollection::Enumerator
+             */
+
+            if (SubStringCount(uid, ".") < 2)
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Metadata/F1KeywordsGenerator.cs]:
```suggestion
            if (uid.Count(c => c == '.') < 2)
```

File: ECMA2Yaml/ECMAHelper/Metadata/F1KeywordsGenerator.cs [hunk 2]
```diff
@@ -44,6 +44,84 @@ private static IEnumerable<string> GetF1Keywords(ReflectionItem item)
...
+                case ItemType.Interface:
+                case ItemType.Struct:
+                case ItemType.Delegate:
+                    break;
+                default:
+                    if (lastindex > 0)
+                    {
+                        var penultimateindex = uid.LastIndexOf(".", lastindex - 1);
+                        if (penultimateindex != -1)
+                            uid= uid.Substring(penultimateindex + 1);
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Metadata/F1KeywordsGenerator.cs]: better not change the value of `uid` here, just use a new variable

File: ECMA2Yaml/ECMAHelper/Metadata/F1KeywordsGenerator.cs [hunk 3]
```diff
@@ -44,6 +44,84 @@ private static IEnumerable<string> GetF1Keywords(ReflectionItem item)
...
+            }
+        }
+
+        /// <summary>
+        /// Count the number of substring in a string.
+        /// </summary>
+        /// <param name="str"> string </param>
+        /// <param name="subString">sub string</param>
+        /// <returns>return number of occurrences</returns>
+        private static int SubStringCount(string str, string subString)
```
Comment (TianqiZhang) [inline ECMA2Yaml/ECMAHelper/Metadata/F1KeywordsGenerator.cs]: no need for this method

File: ECMA2Yaml/ECMAHelper/Metadata/F1KeywordsGenerator.cs (hunk 1 above)
Comment (v-haiboz) [reply ECMA2Yaml/ECMAHelper/Metadata/F1KeywordsGenerator.cs]: fixed

File: ECMA2Yaml/ECMAHelper/Metadata/F1KeywordsGenerator.cs (hunk 3 above)
Comment (v-haiboz) [reply ECMA2Yaml/ECMAHelper/Metadata/F1KeywordsGenerator.cs]: fixed

Comment (v-haiboz) [reply ECMA2Yaml/ECMAHelper/Metadata/F1KeywordsGenerator.cs]: remove

File: ECMA2Yaml/ECMAHelper/Metadata/F1KeywordsGenerator.cs (hunk 2 above)
Comment (v-haiboz) [reply ECMA2Yaml/ECMAHelper/Metadata/F1KeywordsGenerator.cs]: fixed, redefine a new string.

**Review by TianqiZhang — APPROVED** [review]

**Pull request merged**